### PR TITLE
allow specifying a user agent for azure clients

### PR DESCRIPTION
**Pull request by marwanad**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

Those clients are used by cluster-autoscaler. The user agent will be defaulted to `Go/go1.14.4 (amd64-linux) go-autorest/v13.4.0; kubernetes-cloudprovider/v` which is misleading.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/area provider/azure
/sig cloud-provider
/priority important-soon

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/clients/containerserviceclient/azure_containerserviceclient.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ type Client struct {
 func New(config *azclients.ClientConfig) *Client {
 	baseURI := config.ResourceManagerEndpoint
 	authorizer := config.Authorizer
-	armClient := armclient.New(authorizer, baseURI, "", APIVersion, config.Location, config.Backoff)
+	armClient := armclient.New(authorizer, baseURI, config.UserAgent, APIVersion, config.Location, config.Backoff)
```
Comment (marwanad) [inline staging/src/k8s.io/legacy-cloud-providers/azure/clients/containerserviceclient/azure_containerserviceclient.go:60]: armClient will populate if empty:

```
	if userAgent == "" {
		restClient.UserAgent = GetUserAgent(restClient)
	}
```

Comment (feiskyer) [reply staging/src/k8s.io/legacy-cloud-providers/azure/clients/containerserviceclient/azure_containerserviceclient.go:60]: looks good.

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve
/milestone v1.19
/retest

**Comment by marwanad** [comment]
one last one:

/retest

**Pull request merged**